Server/mods/deathmatch/logic/packets/CVehicleInOutPacket.cpp

```diff
@@ -22,9 +23,10 @@ CVehicleInOutPacket::CVehicleInOutPacket()
     m_ucDoor = 0;
 }
 
-CVehicleInOutPacket::CVehicleInOutPacket(ElementID ID, unsigned char ucSeat, unsigned char ucAction)
+CVehicleInOutPacket::CVehicleInOutPacket(ElementID PedID, ElementID VehicleID, unsigned char ucSeat, unsigned char ucAction)
```

Comment:
The "packet source element" is not used anywhere from what I can tell? unless net is doing something. It's a thing in CPacket that CVehicleInOutPacket is based on.

So because it is seemingly not used, I removed it so it doesnt cause confusion. It might have made sense when enter/exit only worked for players, but now the player who initiates the packet reply isn't necessarily the ped that the packet is about. Hope that makes sense.

Reply:
Source player is used for the `source` variable in, for example, trigger event.
Theres another variable `client` (the socket from which the packet came from).

Reply:
Are we talking about the same thing here?
I mean it is not used in CVehicleInOutPacket (specifically) and later when the packet is broadcasted.

Reply:
I was suggesting that the packet source element could be the ped that the packet is about. If the packet source element cannot be a ped (and must be a player), then we can scrap this suggestion.

Is it possible for you to still provide code comments about each of the overloads?

Reply:
Yeah it has to be a player. I think the confusion arrives from this:

**Receiving**
When we receive, the packet source is the client who sent it.
We used the packet source to find who the packet is about (this was the old way).
If we receive packet from old client, ped ID will be him.
If we receive packet from new client, ped ID will be part of packet. (because it can be a ped now)

**Sending**
When we send a packet, the packet source doesn't matter.
We write the player/ped ID to the packet.
(This is unchanged)

So the server->client packet hasn't changed.
Only the client->server packet.

As for CVehicleInOutPacket, I have added an explanation to each of the parameters in the header file.
I don't think it makes much sense to explain the overloads tbh?

We should rather add an explanation to each of the enums in CGame eg. VEHICLE_REQUEST_IN_CONFIRMED and similarly in CClientGame. These took some time to figure out because it's a bounce that goes 
client( CAN I ENTER? )->server( AIGHT )->client( AIGHT I AM INSIDE )->server( AIGHT YOU ARE INSIDE )->client